src/arrow_parquet/uri_utils.rs

```diff
@@ -92,48 +88,40 @@ fn object_store_with_location(uri: &Url, copy_from: bool) -> (Arc<dyn ObjectStor
 
-    let is_test_running = std::env::var("PG_PARQUET_TEST").is_ok();
-
-    if is_test_running {
-        // use minio for testing
-        aws_s3_builder = aws_s3_builder.with_endpoint("http://localhost:9000");
-        aws_s3_builder = aws_s3_builder.with_allow_http(true);
-    }
+    // AWS_ALLOW_HTTP
+    if let Ok(aws_allow_http) = std::env::var("AWS_ALLOW_HTTP") {
+        aws_s3_builder = aws_s3_builder
+            .with_allow_http(aws_allow_http.parse().unwrap_or_else(|e| panic!("{}", e)));
+    };
 
-    let aws_profile_name = std::env::var("AWS_PROFILE").unwrap_or("default".to_string());
+    // first tries to load the profile files from the environment variables and then from the profile
+    let sdk_config = aws_config::defaults(BehaviorVersion::v2024_03_28())
+        .load()
+        .await;
 
-    let region_provider = RegionProviderChain::first_try(EnvironmentVariableRegionProvider::new())
-        .or_else(
-            ProfileFileRegionProvider::builder()
-                .profile_name(aws_profile_name.clone())
-                .build(),
-        );
+    if let Some(credential_provider) = sdk_config.credentials_provider() {
+        if let Ok(credentials) = credential_provider.provide_credentials().await {
+            // AWS_ACCESS_KEY_ID
+            aws_s3_builder = aws_s3_builder.with_access_key_id(credentials.access_key_id());
 
-    let region = region_provider.region().await;
+            // AWS_SECRET_ACCESS_KEY
+            aws_s3_builder = aws_s3_builder.with_secret_access_key(credentials.secret_access_key());
 
-    if let Some(region) = region {
-        aws_s3_builder = aws_s3_builder.with_region(region.to_string());
+            if let Some(token) = credentials.session_token() {
+                // AWS_SESSION_TOKEN
+                aws_s3_builder = aws_s3_builder.with_token(token);
+            }
+        }
     }
 
-    let credential_provider = CredentialsProviderChain::first_try(
-        "Environment",
-        EnvironmentVariableCredentialsProvider::new(),
-    )
-    .or_else(
-        "Profile",
-        ProfileFileCredentialsProvider::builder()
-            .profile_name(aws_profile_name)
-            .build(),
-    );
-
-    if let Ok(credentials) = credential_provider.provide_credentials().await {
-        aws_s3_builder = aws_s3_builder.with_access_key_id(credentials.access_key_id());
-
-        aws_s3_builder = aws_s3_builder.with_secret_access_key(credentials.secret_access_key());
+    // AWS_ENDPOINT_URL
+    if let Some(aws_endpoint_url) = sdk_config.endpoint_url() {
+        aws_s3_builder = aws_s3_builder.with_endpoint(aws_endpoint_url);
+    }
 
-        if let Some(token) = credentials.session_token() {
-            aws_s3_builder = aws_s3_builder.with_token(token);
-        }
+    // AWS_REGION
+    if let Some(aws_region) = sdk_config.region() {
+        aws_s3_builder = aws_s3_builder.with_region(aws_region.as_ref());
     }
 
```

Comment:
does this code also cover other paths like using the EC2 metadata server to obtain credentials?

Reply:
When all required environment variables are configured to auth via EC2 server, `object_store` supports it. But we do not read some of the environment variables from the config.